app/invocation/invocation_action_card.tsx

```diff
@@ -196,6 +225,47 @@ export default class InvocationActionCardComponent extends React.Component<Props
     );
   }
 
+  handleFileClicked(node: any) {
+    let digest_string = node.digest.hash + "/" + node.digest.sizeBytes;
+    let dirFile = "bytestream://" + this.getCacheAddress() + "/blobs/" + digest_string;
+    if (!node.hasOwnProperty("isExecutable")) {
```

Comment:
is there a cleaner way we can tell if this node is not a directory?

Reply:
you could keep them in separate lists/maps instead of shoving them all in a "children" list. that might make it difficult to say sort them in alphabetical order later

another option would be to wrap them in another object that keeps track of their type when you fetch them

Reply:
i ended up wrapping them in another object